### PR TITLE
Add elasticsearch example

### PR DESCRIPTION
**Pull request by alxpy**

### CODE REVIEW

File: examples/elasticsearch/vagga.yaml [hunk 1]
```diff
@@ -0,0 +1,27 @@
+containers:
+  elasticsearch:
+    setup:
+    - !UbuntuRelease { version: 15.10 }
```
Comment (tailhook) [inline examples/elasticsearch/vagga.yaml]: This way is deprecated in vagga 0.6. You should use `- !Ubuntu xenial` or `- !Ubuntu wily` if you want 15.10 specifically.

File: examples/elasticsearch/vagga.yaml [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+      - openjdk-8-jdk
+      - openjdk-8-jre-headless
+      - ca-certificates
+      - ca-certificates-java
+    - !Download
+      url: https://download.elastic.co/elasticsearch/release/org/elasticsearch/distribution/zip/elasticsearch/2.3.3/elasticsearch-2.3.3.zip
+      path: /root/elasticsearch-2.3.3.zip
+    - !Sh |
+        update-ca-certificates -f
+        unzip -q /root/elasticsearch-2.3.3.zip
```
Comment (tailhook) [inline examples/elasticsearch/vagga.yaml]: Do you unzip inside the work? It's definitely an antipattern.

File: examples/elasticsearch/vagga.yaml [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+    - !UbuntuUniverse
+    - !Install
+      - openjdk-8-jdk
+      - openjdk-8-jre-headless
+      - ca-certificates
+      - ca-certificates-java
+    - !Sh update-ca-certificates -f
+    - !Tar
+      url: https://download.elastic.co/elasticsearch/release/org/elasticsearch/distribution/tar/elasticsearch/2.3.3/elasticsearch-2.3.3.tar.gz
+      path: /opt
```
Comment (popravich) [inline examples/elasticsearch/vagga.yaml]: There's a shorter way:
```yaml
    # elastic PGP & Repo
    - !AptTrust
      server: pgp.mit.edu
      keys: [D88E42B4]
    - !UbuntuRepo
      url: http://packages.elastic.co/elasticsearch/2.x/debian
      suite: stable
      components: [main]
    - !Install 
      - elasticsearch
```

File: examples/elasticsearch/vagga.yaml [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+    - !EnsureDir /opt/elasticsearch-2.3.3/data
+    - !EnsureDir /opt/elasticsearch-2.3.3/logs
+    - !EnsureDir /opt/elasticsearch-2.3.3/plugins
+    - !EnsureDir /opt/elasticsearch-2.3.3/config/scripts
+    volumes:
+      /opt/elasticsearch-2.3.3/logs: !BindRW /work/es/logs
+      /opt/elasticsearch-2.3.3/plugins: !BindRW /work/es/plugins
+      /opt/elasticsearch-2.3.3/data: !BindRW /work/es/data
```
Comment (popravich) [inline examples/elasticsearch/vagga.yaml]: I think a note should be added here saying:
> If you don't need to keep Elasticsearch state between the runs you can use either `!Tmpfs`, or `!Snapshot` types
to get writable paths"

**Comment by lud4ik** [comment]
Very interesting and useful example!

**Pull request closed without merging**